interceptors/payment_authentication_test.go

```diff
@@ -3,72 +3,214 @@ package interceptors
 		So(w.Code, ShouldEqual, 400)
 	})
 
 	Convey("Invalid user details in context", t, func() {
 		path := fmt.Sprintf("/payments/%s", "1234")
-		req, err := http.NewRequest("Get", path, nil)
+		req, err := http.NewRequest("GET", path, nil)
 		req = mux.SetURLVars(req, map[string]string{"payment_id": "1234"})
 		req.Header.Set("Eric-Identity", "authorised_identity")
 		req.Header.Set("Eric-Identity-Type", "oauth2")
 		req.Header.Set("ERIC-Authorised-User", "test@test.com;test;user")
-		req.Header.Set("ERIC-Authorised-Roles", "notauth2")
+		req.Header.Set("ERIC-Authorised-Roles", "noroles")
 		// The details have to be in a authUserDetails struct, so pass a different struct to fail
 		authUserDetails := models.PaymentResourceData{
 			Status: "test",
 		}
 		ctx := context.WithValue(req.Context(), helpers.UserDetailsKey, authUserDetails)
 		So(err, ShouldBeNil)
 
+		mock := dao.NewMockDAO(mockCtrl)
+		mockPaymentService := createMockPaymentService(mock, cfg)
+		mockPaymentAuthenticationInterceptor := createMockPaymentPaymentAuthenticationInterceptor(&mockPaymentService)
+
 		w := httptest.NewRecorder()
-		test := PaymentAuthenticationInterceptor(GetTestHandler())
+		test := mockPaymentAuthenticationInterceptor.PaymentAuthenticationIntercept(GetTestHandler())
 		test.ServeHTTP(w, req.WithContext(ctx))
 		So(w.Code, ShouldEqual, 500)
 	})
 
 	Convey("No authorised identity", t, func() {
 		path := fmt.Sprintf("/payments/%s", "1234")
-		req, err := http.NewRequest("Get", path, nil)
+		req, err := http.NewRequest("GET", path, nil)
 		req = mux.SetURLVars(req, map[string]string{"payment_id": "1234"})
 		req.Header.Set("Eric-Identity", "authorised_identity")
 		req.Header.Set("Eric-Identity-Type", "oauth2")
 		req.Header.Set("ERIC-Authorised-User", "test@test.com;test;user")
-		req.Header.Set("ERIC-Authorised-Roles", "notauth2")
+		req.Header.Set("ERIC-Authorised-Roles", "noroles")
 		// Pass no ID (identity)
+		authUserDetails := models.AuthUserDetails{}
+		ctx := context.WithValue(req.Context(), helpers.UserDetailsKey, authUserDetails)
+		So(err, ShouldBeNil)
+
+		mock := dao.NewMockDAO(mockCtrl)
+		mockPaymentService := createMockPaymentService(mock, cfg)
+		mockPaymentAuthenticationInterceptor := createMockPaymentPaymentAuthenticationInterceptor(&mockPaymentService)
+
+		w := httptest.NewRecorder()
+		test := mockPaymentAuthenticationInterceptor.PaymentAuthenticationIntercept(GetTestHandler())
+		test.ServeHTTP(w, req.WithContext(ctx))
+		So(w.Code, ShouldEqual, 401)
+	})
+
+	Convey("Happy path where user is creator", t, func() {
+		path := fmt.Sprintf("/payments/%s", "1234")
+		req, err := http.NewRequest("GET", path, nil)
+		req = mux.SetURLVars(req, map[string]string{"payment_id": "1234"})
+		req.Header.Set("Eric-Identity", "identity")
+		req.Header.Set("Eric-Identity-Type", "oauth2")
+		req.Header.Set("ERIC-Authorised-User", "test@test.com;test;user")
+		req.Header.Set("ERIC-Authorised-Roles", "noroles")
+		// Pass no ID (identity)
```

Comment:
Is this correct?  It's a happy day test so shouldn't it have the identity set - just comment needs updating I'd say